src/pages/Governance/Stake/Index.tsx

```diff
@@ -38,68 +43,133 @@ export function StakePage() {
   };
 
   const onSubmitClick = async () => {
-    // TODO: to be implemented
-    console.log("Submit Staking");
+    if (!stakingAmount) return;
+    await submitStake(parseInt(stakingAmount), operatorAddr, voterAddr);
```

Comment:
How come the stakingAmount is stored as a string and not `number | null`? Keep in mind that parseInt can return `NaN` if the string isn't valid. I would suggest handling the input validation and conversion to integer in `onStakingAmountChange`.

Reply:
we can't set a HTML input to null (getting console error) and we can't set it to undefined as then we get a changing an uncontrolled input to be controlled error.
I added a validation to replace any non-digit to un empty string and then parse the value - can go back to it later if you feel we should do it differently.